### ISSUE TITLE
Cannot find Entity Culling

### ISSUE THREAD

**Issue by itsmeboy0** [Senth/minecraft-mod-manager#56]
Two problems:

I am trying to install this mod called [Entity Culling Fabric/Forge](https://www.curseforge.com/minecraft/mc-mods/entityculling) using `minecraft-mod-manager install entityculling` but it doesn't seem to find it:
```
entityculling
    🔺 Mod not found on any site...
Mod entityculling not found on any site.
Check so that it's name is correct. Or you can set the name by running:
minecraft-mod-manager configure entityculling=NEW_NAME
```
Trying `minecraft-mod-manager install curse:entityculling` doesn't seem to work either:
```
entityculling
    🔺 Mod not found on any site...
Mod entityculling not found on curse.
Check so that it's name is correct. Or you can set the name by running:
minecraft-mod-manager configure entityculling=NEW_NAME
```
Note that there is another [similar mod](https://www.curseforge.com/minecraft/mc-mods/entity-culling) that is forge only but has a dash in between it (`entity-culling`).
However when I download this other mod, it does not appear when doing `minecraft-mod-manager list`, and so I cannot update it.

Am I missing something or doing something wrong?

**Comment by Senth** [comment]
I briefly looked into this since it seemed weird.

Apparently, it's CurseForge's unofficial API that doesn't work as I expected. If I search on `entityculling` it doesn't find it, even though the slug is called `entityculling`. But if I search on `entity culling` it shows up as one of the results... This could be a tricky one to solve... Unless there is an option that I'm missing, which there could be since it's not a documented API.

**Comment by Senth** [comment]
Doesn't seem to be an option I'm missing. Searching for `entityculling` on [CurseForge](https://www.curseforge.com/minecraft/mc-mods) returns the same result (No mods found).

I'd mostly say the responsibility is on CurseForge, but I'll try to fix it somehow.

There is a python library called `compound-word-splitter` which would split the word `entityculling` into `entity culling`. However, it needs `enchant` to be installed on the computer (a third-party library) and only works on Unix-like systems.

**Comment by dorianpb** [comment]
This has been happening to me, too, with [custom entity models](https://www.curseforge.com/minecraft/mc-mods/custom-entity-models-cem/) and [enhanced block entities](https://modrinth.com/mod/ebe), among others. Offering the slug doesn't help.

**Comment by Senth** [comment]
@dorianpb This information (mods that it can't find) will be very helpful with fixing this behavior 🙂 

I have some additional questions that can help me pinpoint the issue (and make sure I fix it). Did you use `mcman install` command, or had you installed them manually and then used `mcman update`?

I'm seeing that enhanced block entities are actually from modrinth, they have a better search engine than CurseForge. Could you supply the exact command you used to install/update enhanced block entities?

**Comment by dorianpb** [comment]
For both of the mods, I initially used `mmm install`, but then tried downloading manually and using `mmm update`. The exact command I used to try downloading enhanced block entites was `mmm install enhancedblockentities=modrinth:ebe`, and for cem it was `mmm install cem=curse:custom-entity-models-cem`

**Comment by Senth** [comment]
Thanks a lot 😃 I have some ideas on how to solve this issue, it's not really a good solution since it uses brute force. I have another solution that requires me to implement and publish a service that splits the single word into multiple ones.

I just also got an idea for displaying some options if there are no matches. For example:

```
No matches found for enhancedblockentities. Does any of these match?
1. enhanced block entities
2. enhanced mobs
3. block entities enhanced
Select mod by entering the number, or just press enter to continue without installing this.
```

**Comment by cobalt2727** [comment]
This still happens for me, did anyone have any ideas?

**Comment by Senth** [comment]
Hi @cobalt2727, this issue hasn't been solved yet. There hasn't been as much development on `mcman` as I hoped. IRL stuff and a bit of coding burnout from having coding as my work as well.

For now, the only thing is to update this manually. I know it's a hassle (since I also have to do this for some mods I personally use), but at least better than having to download all manually.

Since it's a tricky one to solve thanks to curse search is a bit weird.

**Comment by cobalt2727** [comment]
No worries. Thanks for the response!

**Comment by Senth** [comment]
Some updates on this issue.

I found this site (https://www.modpackindex.com/mods) which indexes mods from curse forge. This can be of use since the search here seems to be a lot better than the one at curse. Not sure if this is sufficient though, or if I should create my own indexer service later down the line.

I did find a library that could split words. However, I forgot to comment the library here; hopefully, I can find it again 😅 But I think this will be the next task I'll work on, trying to fix all the search issues.
